code/game/objects/items/sledgehammer.dm

```diff
@@ -0,0 +1,80 @@
+/*
+ * Sledgehammer
+ */
+/obj/item/sledge  // DEM AXES MAN, marker -Agouri
+	icon_state = "hammeroff"
+	item_state = "hammeroff"
+	lefthand_file = 'icons/mob/inhands/weapons/hammers_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/hammers_righthand.dmi'
+	name = "sledgehammer"
+	desc = "Used for knocking the everloving hell out of whatever, or whoever, you hit."
+	force = 5
+	throwforce = 15
+	w_class = WEIGHT_CLASS_BULKY
+	slot_flags = ITEM_SLOT_BACK
+	attack_verb = list("battered", "crushed", "slammed", "hammered")
+	hitsound = 'sound/weapons/smash.ogg'
+	max_integrity = 200
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 30)
+	resistance_flags = FIRE_PROOF
+	var/wielded = FALSE // track wielded status on item
```

Comment:
don't do this, it's poor use of components as you shouldn't need to save the state of the component.